### PR TITLE
curlie: 1.6.9 -> 1.7.1

### PR DESCRIPTION
**Pull request by blackheaven**
###### Description of changes

[Update curlie to 1.7.1](https://github.com/rs/curlie/releases/tag/v1.7.1)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by drupol — CHANGES_REQUESTED** [review]
Looks like something is broken here!

**Comment by blackheaven** [comment]
Anything in particular?

**Comment by drupol** [comment]
The builds are all broken .

File: pkgs/tools/networking/curlie/bump-golang-x-sys.patch [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
- 
+@@ -1,8 +1,8 @@
+ module github.com/rs/curlie
+
  require (
- 	golang.org/x/crypto v0.0.0-20180524125353-159ae71589f3
--	golang.org/x/sys v0.0.0-20191026070338-33540a1f6037
+-	golang.org/x/crypto v0.1.0
+-	golang.org/x/sys v0.1.0
++	golang.org/x/crypto v0.0.0-20180524125353-159ae71589f3
```
Comment (wegank) [inline pkgs/tools/networking/curlie/bump-golang-x-sys.patch:11]: That doesn't make sense under the title `bump-golang-x-sys.patch`. Probably should drop the patch.

Comment (blackheaven) [reply pkgs/tools/networking/curlie/bump-golang-x-sys.patch:11]: it breaks without it

**Comment by blackheaven** [comment]
Let me know if it's okay, so I can squash it

**Comment by wegank** [comment]
> Let me know if it's okay, so I can squash it

Yes, please.

**Comment by blackheaven** [comment]
thanks, done.

**Review by wegank — APPROVED** [review]

**Comment by drupol** [comment]
LGTM

**Pull request merged**